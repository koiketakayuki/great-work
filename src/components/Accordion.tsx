import * as React from 'react';

export class Accordion extends React.Component {
  render() {
    return <div>test</div>;
  }
}
