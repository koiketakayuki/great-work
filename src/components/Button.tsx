import * as React from 'react';
import { ColorType } from '../config/StyleConfig';
import { Paper } from './Paper';
import { Container } from './layout/Container';
import { FloatLevel } from '../lib/getFloatStyle';
import { PaddingProperty } from 'csstype';

export interface ButtonProps {
  padding?: PaddingProperty<string>;
  onClick?: () => void;
  disabled?: boolean;
  float?: FloatLevel;
  type?: ColorType;
}

export class Button extends React.Component<ButtonProps> {

  onClick = () => {
    if (this.props.onClick && !this.props.disabled) {
      this.props.onClick();
    }
  }

  onKeyDown = (e: React.KeyboardEvent) => {
    if (e.key === 'Enter' && this.props.onClick) {
      this.onClick();
    }
  }

  render() {
    return (
      <Paper
        display="inline-block"
        float={this.props.float}
        onClick={this.onClick}
        onKeyDown={this.onKeyDown}
        cursor={this.props.disabled ? 'not-allowed' : 'pointer'}
        type={this.props.disabled ? 'disabled' : (this.props.type || 'primary')}
        tabIndex={0}
      >
        <Container padding={this.props.padding || '8px'}>
          {this.props.children}
        </Container>
      </Paper>
    );
  }
}
