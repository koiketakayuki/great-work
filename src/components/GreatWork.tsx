import * as React from 'react';

export interface HelloProps { name: string; }

export class GreatWork extends React.Component<HelloProps> {
  render() {
    return <h1>GreatWork {this.props.name}!</h1>;
  }
}
