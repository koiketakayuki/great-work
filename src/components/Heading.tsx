import * as React from 'react';
import { Text, TextProps, TextTag } from './Text';
import { StyleConfig } from '../config/StyleConfig';
import { StyleContext } from '../config/StyleContext';

export type HeadingLevel = 1 | 2 | 3 | 4 | 5;

export interface HeadingProps {
  tag?: TextTag;
  level: HeadingLevel;
  children: React.ReactNode;
  config: StyleConfig;
}

export const Heading = (props: HeadingProps) => (
  <StyleContext.Consumer>
    {config => <Text {...(getHeadingStyle(config, props))}>{props.children}</Text>}
  </StyleContext.Consumer>
);

function getHeadingStyle(config: StyleConfig, props: HeadingProps): TextProps {
  if (props.level === 2) {
    return getHeading2Style(config, props);
  }

  if (props.level === 3) {
    return getHeading3Style(config, props);
  }

  if (props.level === 4) {
    return getHeading4Style(config, props);
  }

  if (props.level === 5) {
    return getHeading5Style(config, props);
  }

  return getHeading1Style(config, props);
}

function getHeading1Style(config: StyleConfig, props: HeadingProps): TextProps {
  return {
    tag: props.tag || 'h1',
    fontSize: config.heading1,
    fontWeight: 'bold',
    decoration: 'none',
  };
}

function getHeading2Style(config: StyleConfig, props: HeadingProps): TextProps {
  return {
    tag: props.tag || 'h2',
    fontSize: config.heading2,
    fontWeight: 'bold',
    decoration: 'none',
  };
}

function getHeading3Style(config: StyleConfig, props: HeadingProps): TextProps {
  return {
    tag: props.tag || 'h3',
    fontSize: config.heading3,
    fontWeight: 'bold',
    decoration: 'none',
  };
}

function getHeading4Style(config: StyleConfig, props: HeadingProps): TextProps {
  return {
    tag: props.tag || 'h4',
    fontSize: config.heading4,
    fontWeight: 'bold',
    decoration: 'none',
  };
}

function getHeading5Style(config: StyleConfig, props: HeadingProps): TextProps {
  return {
    tag: props.tag || 'h5',
    fontSize: config.heading5,
    lineHeight: 1.0,
    fontWeight: 'bold',
    decoration: 'none',
    cursor: 'default',
  };
}
