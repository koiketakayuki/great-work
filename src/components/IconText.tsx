import * as React from 'react';
import { Row } from './layout/Row';
import { FixedCell, FlexCell } from './layout/Cell';
import { Icon } from './Icon';
import { Text } from './Text';
import { ColorType } from '../config/StyleConfig';
import { Container } from './layout/Container';
import { CursorProperty } from 'csstype';

export interface IconTextProps {
  icon: string;
  text: string;
  type?: ColorType;
  cursor?: CursorProperty;
  hover?: object;
  onClick?: React.MouseEventHandler;
  onKeyDown?: React.KeyboardEventHandler;
  tabIndex?: number;
  left?: boolean;
}

export const IconText = (props: IconTextProps) => (
  <Row hover={props.hover} cursor={props.cursor} onClick={props.onClick} tabIndex={props.tabIndex} onKeyDown={props.onKeyDown}>
    <FixedCell>
      <Row>
        <Icon name={props.icon} type={props.type}/>
      </Row>
    </FixedCell>
    <FlexCell>
      <Container padding="0 4px">
        <Text type={props.type}>{props.text}</Text>
      </Container>
    </FlexCell>
  </Row>
);
