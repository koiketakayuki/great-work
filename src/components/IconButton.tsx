import * as React from 'react';
import { IconProps } from './Icon';
import { ButtonProps } from './Button';
import { StyleContext } from '../config/StyleContext';
import { StyleConfig } from '../config/StyleConfig';

export interface IconButtonProps extends ButtonProps {
  icon: React.ReactElement<IconProps>;
}

function getIconButton(config: StyleConfig, props: IconButtonProps) {
  const style = {
    color: config.getColor(props.disabled ? 'disabled' : props.type),
    cursor: props.disabled ? 'not-allowed' : 'pointer',
    padding: props.padding,
  };

  const onClick = () => {
    if (props.onClick && !props.disabled) {
      props.onClick();
    }
  };

  return (
    <div onClick={onClick} style={style}>
      {props.icon}
    </div>
  );
}

export function IconButton(props: IconButtonProps) {
  return (
    <StyleContext.Consumer>
      {config => getIconButton(config, props)}
    </StyleContext.Consumer>
  );
}
