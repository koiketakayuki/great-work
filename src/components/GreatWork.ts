import { Button } from './Button';
import { Heading } from './Heading';
import { Icon } from './Icon';
import { IconButton } from './IconButton';
import { IconText } from './IconText';
import { Link } from './Link';
import { Paper } from './Paper';
import { Text } from './Text';
import { CheckList } from './form/CheckList';
import { ErrorMessage } from './form/ErrorMessage';
import { FormContainer } from './form/FormContainer';
import { ImageForm } from './form/ImageForm';
import { FormItem } from './form/FormItem';
import { ListForm } from './form/ListForm';
import { PasswordForm } from './form/PasswordForm';
import { RadioButton } from './form/RadioButton';
import { RadioButtons } from './form/RadioButtons';
import { SelectBox } from './form/SelectBox';
import { TextArea } from './form/TextArea';
import { TextForm } from './form/TextForm';
import { Section } from './layout/Section';

export const GreatWork = {
  Button,
  Heading,
  Icon,
  IconButton,
  IconText,
  Link,
  Paper,
  Text,
  CheckList,
  ErrorMessage,
  FormContainer,
  FormItem,
  ImageForm,
  ListForm,
  PasswordForm,
  RadioButton,
  RadioButtons,
  SelectBox,
  TextArea,
  TextForm,
  Section,
};
