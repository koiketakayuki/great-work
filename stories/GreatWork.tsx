import * as React from 'react';
import { storiesOf } from '@storybook/react';

import { GreatWork } from '../src/components/GreatWork';

storiesOf('Welcome', module)
  .add('to test', () => <GreatWork name="Ryutaro" / >);
