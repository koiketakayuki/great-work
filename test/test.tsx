import * as React from 'react';
import { create } from 'react-test-renderer';
import { GreatWork } from '../src/components/GreatWork';

test('Say my name, say my name...', () => {
  const tree1 = create(
      <GreatWork name="Ryutaro" />,
  ).toJSON();
  expect(tree1).toMatchSnapshot();
});
