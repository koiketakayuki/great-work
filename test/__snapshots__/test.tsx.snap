// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Say my name, say my name... 1`] = `
<h1>
  GreatWork 
  Ryutaro
  !
</h1>
`;
